backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/VdsManager.java

```diff
@@ -1293,11 +1333,40 @@ public void updateV2VJobInfos(List<V2VJobInfo> v2vJobInfos) {
         }
     }
 
+
     private void initAvailableCpus() {
         List<VdsCpuUnit> cpuTopology = cachedVds.getCpuTopology();
-        if (cpuTopology == null) {
+        if (cpuTopology == null || cachedVds.getVdsmCpusAffinity() == null) {
             return;
         }
         this.cpuTopology = cpuTopology;
+        this.cpuTopology.stream().filter(cpu -> cpu.getCpu() == Integer.parseInt(cachedVds.getVdsmCpusAffinity())).forEach(cpu -> cpu.setVmId(Guid.SYSTEM));
+    }
+
+    public List<VdsCpuUnit> getCpuTopology() {
+        List<VdsCpuUnit> clone = new ArrayList<>();
+        cpuTopology.forEach(cpu -> clone.add(cpu.clone()));
+        return clone;
+    }
+
+    public void unpinVmCpus(VM vm) {
```

Comment:
would we unpin the cpus when migrating out a vm with this mechanism or is it something we'll need to handle when adding the dedicated cpus to live migration?

Reply:
The intention was to have the handling of each host cpu topology on it's VdsManager instance. Therefore, I think we should unpin here also on migrating out scenario. Although, I didn't look on this flow yet.